src/sidebar/services/groups.js

```diff
@@ -251,6 +256,34 @@ function groups(
       });
   }
 
+  /*
+   * Clear the state of the groups if there is a direct-linked group
+   * fetch failure or the currently selected group is a private group.
+   *
+   * This is intended to be called during user state changes (aka on
+   * login or logout) so that any events that reference the focused
+   * group wait until the new focused group has been loaded before
+   * executing.
+   *
+   */
+  function clearState() {
```

Comment:
I need help with this name. I wasn't sure how to put all those comments into a short function name.

Reply:
It is getting quite tricky to reason about all the possible states here. I think it would simplify things if we could just always clear all groups when logging out/logging in. Then startup, login and logout will all be the same state.

The reason for not doing that in the past was to avoid losing the focused group on login/logout. Since it is persisted to local storage however, it will be restored from local storage after the login/logout completes, as long as the persisted focused group isn't lost when it briefly becomes `null`. That will require a small change to the logic lower down the file which subscribes to focused group changes.

Reply:
For consistency with the other methods, I would have preferred this method to be named just `clear` - that would only make sense if there weren't a bunch of conditions attached to it though.

Reply:
So as discussed, if we clear the groups on logout/login and there is a direct link it will display the direct linked group/annotation again even though the user may have 'cleared' that state by clicking on another group. I'm going to look into perhaps clearing the direct-linked settings in that case which is more explicit and will hopefully allow us to do as you suggested and simplify this logic.